src/index.js

```diff
@@ -6,46 +6,52 @@ const tags = tagsW3.map(tag => tag.replace(/[<|>]/g, ""));
 
 // Wait for the DOM to be fully loaded
 document.addEventListener("DOMContentLoaded", function () {
-    uploadTagsFromLocalStorage(); // list all tags in localStorage
-    updateProgressBar();
-    // Find the form element and add a submit event listener
-    const form = document.querySelector("form");
-    form.addEventListener("submit", function (event) {
-        // on submit
-        event.preventDefault(); // Prevent the default form submission
-        submitVal(); // call our submitVal function
-    });
 
-    function submitVal() {
-        let val = document.getElementById("input1").value;
-        // remove all white spaces and non-alphabetical characters except . ! - 
-        let regex = /[^a-zA-Z!-.]/g;
-        // change input to lower case and apply regex
-        let sanitizedVal = val.toLowerCase().replace(regex, "");
+  uploadTagsFromLocalStorage(); // list all tags in localStorage
+  updateProgressBar();
+  // Find the form element and add a submit event listener
+  const form = document.querySelector("form");
+  form.addEventListener("submit", function (event) {
+    // on submit
+    event.preventDefault(); // Prevent the default form submission
+    submitVal(); // call our submitVal function
+  });
 
-        let li = "<li>" + sanitizedVal + "</li>";
-        // check if the input value is in the tags array
-        let found = tags.some(tag => tag.toLowerCase() === sanitizedVal);
-        if (found) {
-            document.getElementById("list").insertAdjacentHTML("beforeend", li);
-            saveTagToLocalStorage(sanitizedVal); // Save tag to localStorage
-            updateProgressBar();
-        }
-        document.getElementById("input1").value = "";
+  function submitVal() {
+    let val = document.getElementById("input1").value;
+    // remove all white spaces and non-alphabetical characters except . ! - 
+    let regex = /[^a-zA-Z!-.]/g;
+    // change input to lower case and apply regex
+    let sanitizedVal = val.toLowerCase().replace(regex, "");
+    let li = "<li>" + sanitizedVal + "</li>";
+    // check if the input value is in the tags array
+    let found = tags.some(tag => tag.toLowerCase() === sanitizedVal);
+    // get the list of inputs already made
+    const getList = localStorage.getItem("myTags");
+    let checkDuplicate = false;
+    // ? checks if the list is not empty first, to make includes() work
+    if (getList?.includes(val)) {
```

Comment:
On line 33, the argument of `includes()` has to be `sanitizedVal`. Otherwise, it will allow for duplicates. If `a` is in the list already, and a user types <a>, the function checks <a> against the list from local storage. It's not there since the list is sanitized, so it will add `a` (in sanitized form, because the <li> elements are sanitized) to the list.

Reply:
Oh man! You think you have tested properly, but it seems that it wasn't the case. 😅